src/dtls.c

```diff
@@ -424,4 +424,149 @@ int wolfSSL_dtls_cid_get_tx(WOLFSSL* ssl, unsigned char* buf,
 
 #endif /* WOLFSSL_DTLS_CID */
 
+#if defined(NO_SHA) && defined(NO_SHA256)
+#error "DTLS needs either SHA or SHA-256"
+#endif /* NO_SHA && NO_SHA256 */
+
+#if !defined(NO_SHA) && defined(NO_SHA256)
+#define DTLS_COOKIE_TYPE WC_SHA
+#define DTLS_COOKIE_SZ WC_SHA_DIGEST_SIZE
+#endif /* !NO_SHA && NO_SHA256 */
+
+#ifndef NO_SHA256
+#define DTLS_COOKIE_TYPE WC_SHA256
+#define DTLS_COOKIE_SZ WC_SHA256_DIGEST_SIZE
+#endif /* !NO_SHA256 */
+
+typedef struct Vector {
+    word32 size;
+    const byte *elements;
+} Vector;
+
+typedef struct CH {
+    ProtocolVersion* pv;
+    const byte *random;
+    Vector sessionId;
+    Vector cookie;
+    Vector cipherSuite;
+    Vector compression;
+    Vector extension;
+    word16 length;
+} CH;
```

Comment:
`Vector` and `CH` may collide with other projects/libraries. I think this should be more namespaced.

Reply:
being the struct declared and used only in `dtls.c`, isn't adding a prefix superfluous?

Reply:
I'm not sure how this would work if users include custom/other headers in their `user_settings.h` file. I think that it would be a good idea to add a namespace to these structs due to their very generic names.

Reply:
@rizlik Missed the `struct CH`.